ai/providers/openai.py

```diff
@@ -33,13 +32,27 @@ def get_models(self) -> dict:
     def generate_response(self, prompt: str, system_content: str) -> str:
         try:
             self.client = openai.OpenAI(api_key=self.api_key)
-            response = self.client.chat.completions.create(
+            response = self.client.responses.create(
                 model=self.current_model,
-                n=1,
-                messages=[{"role": "system", "content": system_content}, {"role": "user", "content": prompt}],
-                max_tokens=self.MODELS[self.current_model]["max_tokens"],
+                input=[
+                    {"role": "developer", "content": system_content},
+                    {"role": "user", "content": prompt},
+                ],
+                max_completion_tokens=self.MODELS[self.current_model]["max_tokens"],
             )
-            return response.choices[0].message.content
+            # Prefer the new property if available
+            if hasattr(response, "output_text") and response.output_text:
+                return response.output_text
+
+            # Fallback for older models/SDKs
+            if hasattr(response, "output"):
+                for output in response.output:
+                    if output.get("role") == "assistant":
+                        for content in output.get("content", []):
+                            if content.get("type") == "output_text":
+                                return content.get("text")
+
```

Comment:
```suggestion
```

🪓 Personal opinion, but we do not need to support older SDK versions while using `requirements.txt` in this sample.

Without tests this becomes difficult to maintain as well and I'm finding all of the current models return above, but please let me know if something else is happening for you!